### PR TITLE
chore: Upgrading Metro dependencies to 0.77.0

### PR DESCRIPTION
**Pull request by arushikesarwani94**
Summary:
---------

Updating the versions of Metro in the cli-plugin-metro to incorporate the new Metro Release v0.77.0.
https://github.com/facebook/metro/releases/tag/v0.77.0

Per the release notes shared below, this is a breaking change for RN CLI including bumping minimum Node version to 18, using Terser's default for comments and removing `metro-minify-uglify` from `metro` dependencies

Metro release notes:

* **[Breaking]**: Increase minimum Node version from 16 to 18. (https://github.com/facebook/metro/commit/ebaf27ab49eb8b9394107bd7a796b24bc342c483 by @arushikesarwani94)
* **[Breaking]**: Use Terser's default for comments. (https://github.com/facebook/metro/pull/972 by @tido64)
* **[Breaking]**: Remove `metro-minify-uglify` from `metro` dependencies. (https://github.com/facebook/metro/commit/7a8fcd5ecf5ccbe760c22be7fa924f9cb7c2fe1e by @robhogan)
* **[Fix]**: Fixed a bug in the handling of path separators in `blockList` on Windows. (https://github.com/facebook/metro/commit/2eecfc767095f32bbea35a62c2dc85dcc9c4f839 by @motiz88)
* **[Fix]**: Protect against unsafe merging of `blockList` patterns with inconsistent regex flags. (https://github.com/facebook/metro/commit/c80904b64b8fa5d152da176ce668b77fa3e0ba25 by @motiz88)
* **[Fix]**: Incorrect worker resolution when multiple `metro` versions are installed. (https://github.com/facebook/metro/commit/6d46078e74ae9a43aa90bed46dbd6610e2696cd0 by @robhogan)
* **[Performance]**: Improve resolution performance for files outside the project root. (https://github.com/facebook/metro/commit/dc3cddf21803eb3addb084627e4d536318e3da0a by @robhogan)
* **[Performance]**: Improved startup speed via a new file map cache format. (https://github.com/facebook/metro/commit/166477e9154b3aec4d3081a759cc40f8e3d53c4a by @robhogan)

> NOTE: Experimental features are not covered by semver and can change at any time.

* **[Experimental]**: `metro-file-map`: Normalize root-relative paths for cross-platform cache compatibility. (https://github.com/facebook/metro/commit/d282a0818b840f1d5b0a2086051ec2c7febf1986 by @motiz88)
* **[Experimental]**: Fix crash when `require.context` is used on a directory with infinite path expansions. (https://github.com/facebook/metro/commit/14d652f411fd055c950f820d2de3087e2a791acd by @robhogan)

**Full Changelog**: https://github.com/facebook/metro/compare/v0.76.7...v0.77.0

Reminder: When React Native updates the CLI to a version that depends on metro 0.77.0, it must also update `metro-runtime` etc to 0.77.0 in the same commit.

Test Plan:
----------

✅  yarn build
✅  yarn test

### CODE REVIEW

**Review by szymonrybczak — APPROVED** [review]
Thank you! FYI I re-ran the e2e tests and they are passing now ✅

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [16, 18]
+        node-version: [18]
```
Comment (motiz88) [inline .github/workflows/test.yml:30]: Should this include [Node 20](https://nodejs.org/en/blog/announcements/v20-release-announce) as well? (It's the Current version, will be LTS in October)

Comment (szymonrybczak) [reply .github/workflows/test.yml:30]: +1, I think we can cover this change as part of https://github.com/react-native-community/cli/pull/1986

Comment (arushikesarwani94) [reply .github/workflows/test.yml:30]: https://github.com/react-native-community/cli/pull/2015

**Pull request merged**